### PR TITLE
OBMIN-300

### PR DESCRIPTION
**Pull request by ImAndVlad**
- changed response code to 400 "Bad Request" for advertisements for the selected category
- refacted tests

### CODE REVIEW

File: src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java [hunk 1]
```diff
@@ -110,16 +110,15 @@ public ResponseEntity<Page<AdvertisementTitleDto>> getPageOfAdvertisementsByKeyw
     @Operation(summary = "Find the required number of advertisements for the selected category and return it as a page result")
     @ApiResponses(value = {
             @ApiResponse(responseCode = "200", description = "OK"),
-            @ApiResponse(responseCode = "400", description = "BAD REQUEST"),
-            @ApiResponse(responseCode = "404", description = "NOT FOUND")})
+            @ApiResponse(responseCode = "400", description = "BAD REQUEST")})
     public Page<AdvertisementTitleDto> findAdvertisementsHavingCategory(
             @PathVariable("category_id") Long categoryId,
```
Comment (rpkyrych) [inline src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java:115]: You can annotate that as Positive in addition to verifying the category for existence.

File: src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java [hunk 2]
```diff
@@ -110,16 +110,15 @@ public ResponseEntity<Page<AdvertisementTitleDto>> getPageOfAdvertisementsByKeyw
...
             @Parameter(name = "page", description = "Results page you want to retrieve (0..N). Default value: 0")
             @RequestParam(value = "page", required = false, defaultValue = "0") @PositiveOrZero int page,
             @Parameter(name = "size", description = "Number of records per page. Default value: 12")
-            @RequestParam(value = "size", required = false, defaultValue = "12") @PositiveOrZero int size) throws CategoryIdNotFoundException {
+            @RequestParam(value = "size", required = false, defaultValue = "12") @PositiveOrZero int size) throws BadRequestException {
         if (!categoryService.isCategoryExistsById(categoryId)) {
-            throw new CategoryIdNotFoundException(
+            throw new BadRequestException(
```
Comment (rpkyrych) [inline src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java:121]: I think Not Found was a valid response for categories ID maximum value + 1

File: src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java (hunk 1 above)
Comment (ImAndVlad) [reply src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java]: added

File: src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java (hunk 2 above)
Comment (ImAndVlad) [reply src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java]: Done

**Review by rpkyrych — APPROVED** [review]
Looks good!

**Review by Sershok — APPROVED** [review]

**Review by IllyaZuew — APPROVED** [review]

**Review by demus-n — APPROVED** [review]

**Pull request merged**